### PR TITLE
enhance os image search

### PR DESCRIPTION
**Pull request by tedteng**
**How to categorize this PR?**

/area ops-productivity
/kind enhancement
/platform gcp

**What this PR does / why we need it**:
It is similar to https://github.com/gardener/gardener-extension-provider-aws/issues/713 , a sustainable way to look for available bastion OS images instead of a hard code in bastion, 
**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other operator
a sustainable way to look for available bastion OS images
```

### CODE REVIEW

File: pkg/controller/bastion/actuator_reconcile.go [hunk 1]
```diff
@@ -314,12 +325,33 @@ func disksDefine(opt *Options) []*compute.AttachedDisk {
...
 		Description: "Gardenctl Bastion disk",
 		Name:        diskName,
 		SizeGb:      10,
-		SourceImage: "projects/debian-cloud/global/images/family/debian-10",
+		SourceImage: fmt.Sprintf("projects/%s/global/images/family/%s", osImage, osFamily),
 		Zone:        zone,
 	}
 }
+
+func getosImage(gcpClient gcpclient.Interface) (string, error) {
```
Comment (kon-angelo) [inline pkg/controller/bastion/actuator_reconcile.go:338]: rename to `getOSImage`

Comment (tedteng) [reply pkg/controller/bastion/actuator_reconcile.go]: a2f0b7ec done

**Review by kon-angelo — APPROVED** [review]
/lgtm

**Pull request merged**